src/Arcus.Messaging.Pumps.ServiceBus/Correlation/MessageCorrelationServiceBusPlugin.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Threading.Tasks;
+using Arcus.Messaging.Abstractions;
+using GuardNet;
+using Microsoft.Azure.ServiceBus;
+using Microsoft.Azure.ServiceBus.Core;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Arcus.Messaging.Pumps.ServiceBus.Correlation
+{
+    /// <summary>
+    /// Azure ServiceBus plugin to add correlation information to the inbound and outbound messages.
+    /// </summary>
+    public class MessageCorrelationServiceBusPlugin : ServiceBusPlugin
+    {
+        private readonly ILogger<MessageCorrelationServiceBusPlugin> _logger;
+
+        /// <summary>
+        /// Gets the user property name that will be included in the Service Bus <see cref="Message"/>.
+        /// </summary>
+        public const string CorrelationInfoUserProperty = nameof(MessageCorrelationInfo);
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MessageCorrelationServiceBusPlugin"/> class.
+        /// </summary>
+        /// <param name="serviceProvider"></param>
+        public MessageCorrelationServiceBusPlugin(IServiceProvider serviceProvider)
+        {
+            Guard.NotNull(serviceProvider, nameof(serviceProvider));
+            
+            _logger = serviceProvider.GetRequiredService<ILogger<MessageCorrelationServiceBusPlugin>>();
+        }
+
+        /// <summary>
+        /// Gets the name of the plugin.
+        /// </summary>
+        public override string Name => nameof(MessageCorrelationServiceBusPlugin);
+
+        /// <summary>
+        /// Append the <see cref="MessageCorrelationInfo"/> after the message is received.
+        /// </summary>
+        /// <param name="message">The received message.</param>
+        public override Task<Message> AfterMessageReceive(Message message)
```

Comment:
Now that I see the changes I'm wondering what the benefit is?

It feels more like duplicating the correlation information, no?

Maybe it's better to have an extension that returns `MessageCorrelationInfo` for a given `Message` such as `message.GetCorrelationInfo()`?

Reply:
Yeah, was thinking the same...
I'll change that, OK?

Reply:
Sounds good to me!

Reply:
Done.